### ISSUE TITLE
Improve highest account index restore

### ISSUE THREAD

**Issue by kyranjamie** [blockstack/blockstack-app#4346]
The behaviour of how to figure out how many accounts you have created in the past, relies on a recursive look up process. 

This runs in the page's context, and takes some time, and if the user quits during that operation, nothing happens. Let's improve this so it's more reliable.

One way to do this might be to move the action to the background script so it continues to run once the window is closed.

**Comment by markmhendrickson** [comment]
@kyranjamie relevant to your current key work for mobile?

**Comment by kyranjamie** [comment]
Not really, this is specific to extension in that we could improve highest account index lookup by running it in bg script (and not losing progress if user closes window)

**Comment by 314159265359879** [comment]
Users of mobile to want this functionality there too:
User want option to restore account names with BNS name, instead of manual entry: https://github.com/leather-io/mono/issues/1378
User want option to restore all accounts with a balance automatically: https://github.com/leather-io/mono/issues/1377
